src/main/python/ayab/preferences.py

```diff
@@ -242,16 +267,18 @@ class PrefsLangWidget(QComboBox):
     @author Tom Price
     @date   July 2020
     """
-    def __init__(self, prefs):
+    def __init__(self, prefs:Preferences):
         super().__init__()
         self.prefs = prefs
         self.prefs.languages.add_items(self)
 
-    def connectChange(self):
+    def connectChange(self)->None:
         self.currentIndexChanged.connect(self.update_setting)
 
-    def update_setting(self):
+    def update_setting(self)->None:
         self.prefs.settings.setValue("language", self.currentData())
 
-    def refresh(self):
+    def refresh(self)->None:
         self.setCurrentIndex(self.findData(self.prefs.value("language")))
+
+PrefsWidgetTypes:TypeAlias = PrefsBoolWidget|PrefsLangWidget|PrefsComboWidget
```

Comment:
We _could_ make a root class that all 3 widgets inherit from, but that felt excessive for this use.